### PR TITLE
improved scrollTo function to allow units (new unit is rows)

### PR DESCRIPTION
**Pull request by Falseee**
Feature requested by @ynnckstrm in Issue #3476.

Jsfiddle: http://jsfiddle.net/31qru48k/1/

### CODE REVIEW

File: site/docs/api/methods.md [hunk 1]
```diff
@@ -381,12 +381,15 @@ The calling method syntax: `$('#table').bootstrapTable('method', parameter)`.
...
 
 - **Detail:**
 
-  Scroll to the number `value` position, the unit is `'px'`, set `'bottom'` means scroll to the bottom.
-
+  - value
+    - Scroll to the number `value` position, the unit is `'px'`, set `'bottom'` means scroll to the bottom.
+  - object
+    -  Scroll to the unit (`px` or `rows (index starts by 0)`)   
+    Default: `{unit: 'px', value: 0}`
```
Comment (wenzhixin) [inline site/docs/api/methods.md]: Add a new line

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -3150,16 +3150,32 @@
...
-      if (typeof _value === 'undefined') {
+    scrollTo (params) {
+      if (typeof params === 'undefined') {
         return this.$tableBody.scrollTop()
       }
 
-      let value = _value
-      if (typeof _value === 'string' && _value === 'bottom') {
-        value = this.$tableBody[0].scrollHeight
+      let options = {unit: 'px', value: 0};
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: remove `;`

File: src/bootstrap-table.js [hunk 3]
```diff
@@ -3150,16 +3150,32 @@
...
+      let scrollTo = 0;
+      if (typeof params === 'object') {
+        options = $.extend(options, params);
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: `Object.assign(options, params)`, because we plan to remove jQuery.

File: src/bootstrap-table.js [hunk 4]
```diff
@@ -3150,16 +3150,32 @@
...
+      } else if (typeof params === 'string' && params === 'bottom') {
+        options.value = this.$tableBody[0].scrollHeight
+      } else if (typeof params === 'string') {
+        options.value = params
       }
-      this.$tableBody.scrollTop(value)
+
+      scrollTo = options.value;
+      if(options.unit === "rows") {
```
Comment (wenzhixin) [inline src/bootstrap-table.js]:
```
if (options.unit === 'rows') {
```

File: src/bootstrap-table.js [hunk 5]
```diff
@@ -3150,16 +3150,32 @@
...
+        scrollTo = 0;
+        this.$body.find('> tr').each(function (i) {
+          if (i < options.value) {
+            scrollTo += $(this).height();
+          }
+        });
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: I think it's better to use:
```js
this.$body.find(`> tr:lt(${options.value})`).each((i, el) => {
  scrollTo += $(el).outerHeight(true)
})
```

File: site/docs/api/methods.md (hunk 1 above)
Comment (Falseee) [reply site/docs/api/methods.md]: done

File: src/bootstrap-table.js (hunk 2 above)
Comment (Falseee) [reply src/bootstrap-table.js]: done

File: src/bootstrap-table.js (hunk 3 above)
Comment (Falseee) [reply src/bootstrap-table.js]: Good to know, I try not to use jquery next time.

File: src/bootstrap-table.js (hunk 4 above)
Comment (Falseee) [reply src/bootstrap-table.js]: done

File: src/bootstrap-table.js (hunk 5 above)
Comment (Falseee) [reply src/bootstrap-table.js]: Didnt knew that this selector exists, thank you !

File: src/bootstrap-table.js [hunk 6]
```diff
@@ -3150,16 +3150,30 @@
...
+    scrollTo (params) {
+      if (typeof params === 'undefined') {
         return this.$tableBody.scrollTop()
       }
 
-      let value = _value
-      if (typeof _value === 'string' && _value === 'bottom') {
-        value = this.$tableBody[0].scrollHeight
+      let options = {unit: 'px', value: 0}
+      let scrollTo = 0;
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: Other `;`, can use `yarn lint --fix`

Comment (wenzhixin) [reply src/bootstrap-table.js]: can remove this line

File: src/bootstrap-table.js [hunk 7]
```diff
@@ -3150,16 +3150,30 @@
...
+      if (typeof params === 'object') {
+        options = Object.assign(options, params);
+      } else if (typeof params === 'string' && params === 'bottom') {
+        options.value = this.$tableBody[0].scrollHeight
+      } else if (typeof params === 'string') {
+        options.value = params
       }
-      this.$tableBody.scrollTop(value)
+
+      scrollTo = options.value;
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: define here:
`let scrollTo = options.value`

File: src/bootstrap-table.js (hunk 6 above)
Comment (Falseee) [reply src/bootstrap-table.js]: Never used lint before, thank your for the tip.

File: src/bootstrap-table.js (hunk 7 above)
Comment (Falseee) [reply src/bootstrap-table.js]: done

**Comment by wenzhixin** [comment]
Thanks, I will add a `lint` Travis to check the status.

**Pull request merged**